### DISCUSSION TITLE
The `syncWithoutDetaching` method can cause excessive RAM consumption in large databases

### DISCUSSION THREAD

**Discussion by MohammadZarifiyan** [laravel/framework#53556] [Q&A]
### Laravel Version

11.32

### PHP Version

8.1

### Database Driver & Version

MySQL 8.0.31

### Description

Using `syncWithoutDetaching` will cause a shortage of RAM in the system.

### Steps To Reproduce

First create `post_user` pivot table and add 10 million rows to it. The run the following code:
```php
$user = \App\Models\User::first();
$post = \App\Models\Post::first();

$user->posts()->syncWithoutDetaching([
  $post->getKey() => ['status' => 'pending']
]);
```

**Comment by ghost** [comment]
I would suggest to use mass update instead of this for this case. Bear in mind it will not trigger events like that. So in conclusion I don't think this is a bug/issue.

**Reply by MohammadZarifiyan** [reply]
Laravel receives the rows in the pivot all at once and performs comparison operations. This is what causes the problem.
